### PR TITLE
Bump gRPC to v1.47.5

### PR DESCRIPTION
**Pull request by rschu1ze**
See #55916

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 94bfa489904a3eb9a1d7ec705520e2c752726de0 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56059/94bfa489904a3eb9a1d7ec705520e2c752726de0/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6659351807">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56059/94bfa489904a3eb9a1d7ec705520e2c752726de0/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by Algunenano — CHANGES_REQUESTED** [review]
GRPC seems to make releases once a month or so. I don't mind if you want to test each one but let's not make another 15 PRs. We can test them incrementally in this PR instead.

**Comment by rschu1ze** [comment]
Okay, sounds like a plan.

**Comment by rschu1ze** [comment]
@Felixoid This PR updates a submodule and CI jobs [BuilderDebDebug](https://github.com/ClickHouse/ClickHouse/actions/runs/6789619247/job/18466282149?pr=56059) and [BuilderBinDarwinAarch64](https://github.com/ClickHouse/ClickHouse/actions/runs/6789619247/job/18466281592?pr=56059) fail for a reason that I don't understand. Do you know what is going wrong there?

**Comment by Felixoid** [comment]
The output is a bit confusing, yes. It basically says, that the build already failed once in https://s3.amazonaws.com/clickhouse-builds/PRs/56059/85c158894126a840f65b46ebe07f979e8c091193/package_debug/build_log.log

Since you are here, can you add https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/build_check.py#L354-L358 to https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/build_check.py#L169, please? It will help to avoid confusing

**Comment by Felixoid** [comment]
See the report https://s3.amazonaws.com/clickhouse-test-reports/56059/85c158894126a840f65b46ebe07f979e8c091193/clickhouse_build_check/report.html, all failed tasks have sparse checkout. I think you need to adjust the sparse configuration for the submodule

**Comment by rschu1ze** [comment]
Okay, thanks, I did as you said.

**Comment by rschu1ze** [comment]
> I don't mind if you want to test each one but let's not make another 15 PRs. We can test them incrementally in this PR instead.

@Algunenano

Since a bit of extra stuff accumulated in this PR and we jump 5 versions, do you mind approving so we can make a "checkpoint"? Of course, I'd batch multiple updates in a future PR again.

File: contrib/sparse-checkout/update-grpc.sh [hunk 1]
```diff
@@ -6,12 +6,12 @@ FILES_TO_CHECKOUT=$(git rev-parse --git-dir)/info/sparse-checkout
 echo '/*' > $FILES_TO_CHECKOUT
 echo '!/test/*' >> $FILES_TO_CHECKOUT
 echo '/test/build/*' >> $FILES_TO_CHECKOUT
+echo '/test/core/tsi/alts/fake_handshaker/*' >> $FILES_TO_CHECKOUT
```
Comment (Algunenano) [inline contrib/sparse-checkout/update-grpc.sh:9]: Why do we need these files (both test/build and test/core)? Are they needed for the build to succeed?

**Review by Algunenano — APPROVED** [review]

File: contrib/sparse-checkout/update-grpc.sh (hunk 1 above)
Comment (rschu1ze) [reply contrib/sparse-checkout/update-grpc.sh:9]: Yes, gRPC moved some files around with the result that we now need to check them out (even if it looks silly).

**Pull request merged**